src/sonic_ax_impl/mibs/__init__.py

```diff
@@ -538,11 +544,25 @@ class Namespace:
     @staticmethod
     def init_namespace_dbs():
         db_conn = []
-        SonicDBConfig.load_sonic_global_db_config()
-        for namespace in SonicDBConfig.get_ns_list():
-            db = SonicV2Connector(use_unix_socket_path=True, namespace=namespace, decode_responses=True)
+        if not SonicDBConfig.isInit():
+            if multi_asic.is_multi_asic():
+                SonicDBConfig.load_sonic_global_db_config()
+            else:
+                SonicDBConfig.load_sonic_db_config()
+        host_namespace_idx = 0
+        ns_list = list(SonicDBConfig.get_ns_list())
+        for namespace in ns_list:
+            if namespace == multi_asic.DEFAULT_NAMESPACE:
+                host_namespace_idx = ns_list.index(namespace)
+            db = SonicV2Connector(use_unix_socket_path=True, namespace=namespace)
             db_conn.append(db)
-
+        # Ensure that db connector of default namespace is the first element of
+        # db_conn list.
+        if host_namespace_idx != 0:
+            tmp_swap = db_conn[0]
+            db_conn[0] = db_conn[host_namespace_idx]
+            db_conn[host_namespace_idx] = tmp_swap
```

Comment:
```suggestion
        db_conn[0], db_conn[host_namespace_idx] = db_conn[host_namespace_idx], db_conn[0]
```
You even do not need to check with 0.